src/main/java/uk/nhs/hee/trainee/details/api/TraineeProfileResource.java

```diff
@@ -144,4 +145,21 @@ public ResponseEntity<String> getVerifiedTraineeIds(@NotNull @RequestParam Strin
       return ResponseEntity.ok(traineeIds.get(0));
     }
   }
+
+  /**
+   * Get the trainee email for a trainee TIS ID.
+   *
+   * @param tisId The TIS ID to search by.
+   * @return The matching trainee email.
+   */
+  @GetMapping("/trainee-email/{tisId}")
+  public ResponseEntity<String> getTraineeEmail(@PathVariable String tisId) {
+    Optional<String> email = service.getTraineeEmailByTisId(tisId);
+
+    if (email.isEmpty()) {
+      return ResponseEntity.notFound().build();
+    } else {
```

Comment:
This conditional is redundant with the optional.
`ResponseEntity.of(optional)` will return either `OK` or `NOT_FOUND` depending on the optional being present.

Reply:
Ah, that's neat.